internal/aws/session.go

```diff
@@ -32,11 +33,14 @@ func NewSession(awsconfig *aws.Config, AWSDebug bool, mc metric.Collector) *sess
 		}
 	})
 
+	notFoundRegex := regexp.MustCompile("^[A-za-z]+NotFound")
 	session.Handlers.Complete.PushFront(func(r *request.Request) {
 		if r.Error != nil {
-			mc.IncAPIErrorCount(prometheus.Labels{"service": r.ClientInfo.ServiceName, "operation": r.Operation.Name})
-			if AWSDebug {
-				glog.ErrorDepth(4, fmt.Sprintf("Failed request: %s/%s, Payload: %s, Error: %s", r.ClientInfo.ServiceName, r.Operation.Name, log.Prettify(r.Params), r.Error))
+			if value, ok := r.Context().Value("report-not-found-error").(bool); !notFoundRegex.MatchString(r.Error.Error()) && !ok || value {
+				mc.IncAPIErrorCount(prometheus.Labels{"service": r.ClientInfo.ServiceName, "operation": r.Operation.Name})
+				if AWSDebug {
+					glog.ErrorDepth(4, fmt.Sprintf("Failed request: %s/%s, Payload: %s, Error: %s", r.ClientInfo.ServiceName, r.Operation.Name, log.Prettify(r.Params), r.Error))
+				}
```

Comment:
This is an odd bit. Let's say that there really is an issue creating an ALB, the error wouldn't properly be sent out unless `AWSDebug` was true, which is not optimal - if something is broken or mis-configured then a user would want to know right away.

Reply:
Like I said below, if this is an actual error, it will surface in the function that actually initiated the API call. Also, the `if AWSDebug { ... }` was there before my change.